cppcache/src/ThinClientPoolDM.cpp

```diff
@@ -2488,9 +2423,8 @@ GfErrType FunctionExecution::execute(void) {
           m_poolDM->getConnectionManager().getCacheImpl()->createDataOutput()),
       funcName, m_args, m_getResult, m_poolDM, m_timeout);
   TcrMessageReply reply(true, m_poolDM);
-  ChunkedFunctionExecutionResponse* resultProcessor(
-      new ChunkedFunctionExecutionResponse(reply, (m_getResult & 2) == 2, *m_rc,
-                                           m_resultCollectorLock));
+  auto* resultProcessor(new ChunkedFunctionExecutionResponse(
+      reply, (m_getResult & 2) == 2, *m_rc, m_resultCollectorLock));
```

Comment:
Admittedly this doesn't have much to do with your change, but any idea why resultProcessor is allocated on the heap?  Also, feel free to ignore as it doesn't pertain to your commit.

Reply:
There are several things wrong with this line.
1) `auto*` really isn't a thing. 
2) a `new` doesn't need to be "copied" into a pointer.
3) Since the `TcrChunkedResult` doesn't really survive outside the life of this method or the reply object it should probably be a `std::shared_ptr` owned by both this method and the reply object or maybe even a `std::unique_ptr` owned by either the method or the reply object. Either way it removes the explicit deletes in multiple places in this method and nothing can leak.

Simple fix:
```C++
  auto resultProcessor = std::unique_ptr<ChunkedFunctionExecutionResponse>(
      new ChunkedFunctionExecutionResponse(reply, (m_getResult & 2) == 2, *m_rc,
                                           m_resultCollectorLock));
  reply.setChunkedResultHandler(resultProcessor.get());
```
and delete all:
```C++
  delete resultProcessor;
  resultProcessor = nullptr;
```